src/pages/settings/Subscription/CardAuthenticationModal/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Modal from '@components/Modal';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import * as PaymentMethods from '@userActions/PaymentMethods';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+
+type CardAuthenticationModalProps = {
+    /** Title shown in the header of the modal */
+    headerTitle?: string;
+};
+function CardAuthenticationModal({headerTitle}: CardAuthenticationModalProps) {
+    const styles = useThemeStyles();
+    const [authenticationLink] = useOnyx(ONYXKEYS.VERIFY_3DS_SUBSCRIPTION);
+    const [session] = useOnyx(ONYXKEYS.SESSION);
+    const [privateStripeCustomerID] = useOnyx(ONYXKEYS.NVP_PRIVATE_STRIPE_CUSTOMER_ID);
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        if (privateStripeCustomerID?.status !== CONST.STRIPE_GBP_AUTH_STATUSES.SUCCEEDED) {
+            return;
+        }
+        PaymentMethods.clearPaymentCard3dsVerification();
+        Navigation.navigate(ROUTES.SETTINGS_SUBSCRIPTION);
+    }, [privateStripeCustomerID]);
+
+    const handleGBPAuthentication = useCallback(
+        (event: MessageEvent<string>) => {
+            const message = event.data;
+            if (message === CONST.GBP_AUTHENTICATION_COMPLETE) {
+                PaymentMethods.verifySetupIntent(session?.accountID ?? -1, true);
+            }
+        },
+        [session?.accountID],
+    );
+
+    useEffect(() => {
+        window.addEventListener('message', handleGBPAuthentication);
+        return () => {
+            window.removeEventListener('message', handleGBPAuthentication);
```

Comment:
This change caused [this bug](https://github.com/Expensify/App/pull/49411#issuecomment-2377694997)

Reply:
This component we were not rendering for the native, so while testing in mobile it wasn't crash for me.

Reply:
@Pujan92 

> This component we were not rendering for the native

Why no? We don't have native.tsx file

Reply:
Ahh I see. Thanks

Reply:
My bad 😄

Reply:
I mean it is from the place where we render this component, in this PR `PaymentCard` component is responsible for calling this and there we have this platform specific file. And if I remember correctly, in the doc it is mentioned that stripe flow won't be allowed in native.